src/main/kotlin/io/github/anki/anki/service/UserService.kt

```diff
@@ -0,0 +1,56 @@
+package io.github.anki.anki.service
+
+import io.github.anki.anki.repository.mongodb.UserRepository
+import io.github.anki.anki.repository.mongodb.document.MongoUser
+import io.github.anki.anki.service.exceptions.UserAlreadyExistException
+import io.github.anki.anki.service.exceptions.UserDoesNotExistException
+import io.github.anki.anki.service.model.User
+import io.github.anki.anki.service.model.mapper.toMongoUser
+import io.github.anki.anki.service.model.mapper.toUser
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.dao.DuplicateKeyException
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.core.userdetails.UsernameNotFoundException
+import org.springframework.stereotype.Service
+
+@Service
+class UserService @Autowired constructor(
+    var userRepository: UserRepository,
+) : UserDetailsService {
+
+    @Throws(UsernameNotFoundException::class)
+    override fun loadUserByUsername(userName: String): UserDetails {
```

Comment:
1) rename to getUserByUserName
2) Throws needs only if u call kotlin code from java code